README.md
## blockscore-php [![Travis Build Status](https://travis-ci.org/BlockScore/blockscore-php.svg?branch=master)](https://travis-ci.org/BlockScore/blockscore-php) [![PHP version](https://badge.fury.io/ph/blockscore%2Fblockscore-php.svg)](https://github.com/BlockScore/blockscore-php/releases)

This is the official PHP client library of the [BlockScore API](https://blockscore.com). You can sign up for an account on [our dashboard](https://manage.blockscore.com/signup).

### Requirements

PHP 5.3.29 and later.

### Composer

This library [can be used through Composer](https://packagist.org/packages/blockscore/blockscore-php). To get started, add the following to your `composer.json`:

```json
{
  "require": {
    "blockscore/blockscore-php": "4.*"
  }
}
```

Make sure to install the changes:

```
composer.phar install
```

Make sure that Composer autoload is added somewhere in your project like so:

```
require_once('vendor/autoload.php');
```

### Manual Installation

If you would rather manually include the library in your project, you can [download this repository](https://github.com/BlockScore/blockscore-php/archive/master.zip) and add the following line to your project:

```
require_once('/path/to/blockscore-php/init.php');
```

### Getting Started

This is an example of creating a person verification and setting up the library:

```php
\BlockScore\BlockScore::setApiKey('sk_test_11111111111111111111111111111111');

$person = \BlockScore\Person::create(array(
    'name_first' => 'Jane',
    'name_last' => 'Doe',
    'birth_day' => 1,
    'birth_month' => 1,
    'birth_year' => 1990,
    'document_type' => 'ssn',
    'document_value' => '0000',
    'address_street1' => '123 Something Ave',
    'address_city' => 'Newton Falls',
    'address_subdivision' => 'OH',
    'address_postal_code' => '44444',
    'address_country_code' => 'US',
));

var_dump($person);
```

### Documentation

Our [API Reference](http://docs.blockscore.com/php) contains all of the code examples for this library.

### Tests

If you want to run our test suite, please install [PHPUnit](https://packagist.org/packages/phpunit/phpunit) using [Composer](https://getcomposer.org/)

```
composer.phar update --dev
```

To run the client lib's test suite, run

```
./vendor/bin/phpunit
```

### Credit

This library is largely based on the wonderful [stripe-php](https://github.com/stripe/stripe-php) library.


CHANGELOG.md
## Changelog

### 4.1.0 - 4 September, 2018
- Added toArray() method for access to raw values.
- Changed Object class for PHP 7.2 compatibility.

### 4.0.3 - 24 March, 2016
- Support time_limit parameter for creating question sets.

### 4.0.2 - 24 March, 2016
- Fixed question sets result scoring.

### 4.0.1 - 11 March, 2015
- Added compatibility for PHP 5.3.29.
- Fixed certain build errors with TravisCI.

### 4.0.0 - 11 March, 2015

- First version of new official BlockScore PHP lib
- Completely overhauled API for making calls to BlockScore
- Added to composer
- Namespaced functions
- Full test coverage
- Improved error handling to use native PHP objects


examples.php
<?php
require_once('blockscore.class.php');

$blockscore = new blockscore('sk_test_n5049aa6053c9a0217bea78070fbf501');

// International Verification (new in API v4: both US & Int'l use the same function)
$name['first'] = 'John';
$name['middle'] = 'W';
$name['last'] = 'Smith';
$dob = '1980-10-10';
$documentNum = 'X110000';
$documentType = 'passport'; //accepted values vary by country, see http://docs.blockscore.com/v4.0/curl/#documents
$address['street1'] = 'Bahnhofstrasse 70';
$address['street2'] = '';
$address['city'] = 'Zurich';
$address['state'] = 'ZH';
$address['postal_code'] = '8001';
$address['country_code'] = 'CH';
// Optional values
$telephone = '+41-22-767-0000';
$ip = 'FE80:0000:0000:0000:0202:B3FF:FE1E:8329';  //or $ip = '127.0.0.1';
$note = 'Test user at ' . date("Y-m-d H:i:s") . ' on ' . gethostname();

try {
	$VerifyIntlResult = $blockscore->Blockscore_Verification($name, $dob, $documentNum, $documentType, $address, $telephone, $ip, $note);
} catch (Exception $e) {
	die( 'Caught exception: ' .  $e->getMessage() . "\n" );
}

echo "VerifyIntlResult: <xmp>";
var_dump($VerifyIntlResult);
echo "</xmp>\n";


// US Verification (new in API v4: both US & Int'l use the same function)
$name['first'] = 'John';
$name['middle'] = 'W';
$name['last'] = 'Smith';
$dob = '1980-10-10';
$documentNum = 'A880000';
$documentType = 'drivers_license'; //accepted values vary by country, see http://docs.blockscore.com/v4.0/curl/#documents
$address['street1'] = '123 Broadway Ave';
$address['street2'] = '';
$address['city'] = 'New York';
$address['state'] = 'NY';
$address['postal_code'] = '10011';
$address['country_code'] = 'US';
// Optional values
$telephone = '+1-212-555-1234';
$ip = 'FE80:0000:0000:0000:0202:B3FF:FE1E:8329';  //or $ip = '127.0.0.1';
$note = 'Test user at ' . date("Y-m-d H:i:s") . ' on ' . gethostname();

try {
	$VerifyUSResult = $blockscore->Blockscore_Verification($name, $dob, $documentNum, $documentType, $address, $telephone, $ip, $note);
} catch (Exception $e) {
	die( 'Caught exception: ' .  $e->getMessage() . "\n" );
}

echo "VerifyUSResult: <xmp>";
var_dump($VerifyUSResult);
echo "</xmp>\n";


// Question Set Retrieval
if(!empty($VerifyUSResult['id']))
{
	try {
		$QuestionSet = $blockscore->QuestionSet();
	} catch (Exception $e) {
		die( 'Caught exception: ' .  $e->getMessage() . "\n" );
	}
	
	echo "QuestionSetResult: <xmp>";
	var_dump($QuestionSet);
	echo "</xmp>\n";
	
	// Question Set Answer Checking
	if(!empty($QuestionSet['id']))
	{
		try {
			$QuestionResults = $blockscore->CheckQuestionAnswers(array(array('question_id'=>1, 'answer_id'=>rand(1,5)),array('question_id'=>2, 'answer_id'=>rand(1,5)),array('question_id'=>3, 'answer_id'=>rand(1,5)),array('question_id'=>4, 'answer_id'=>rand(1,5)),array('question_id'=>5, 'answer_id'=>rand(1,5))));
		} catch (Exception $e) {
				die( 'Caught exception: ' .  $e->getMessage() . "\n" );
		}
		
		echo "Answer Checking Result: <xmp>";
		var_dump($QuestionResults);
		echo "</xmp>\n";
		
		try {
			$QuestionSetRetrieve = $blockscore->RetrieveQuestionSet($QuestionSet['id']);
		} catch (Exception $e) {
			die( 'Caught exception: ' . $e->getMessage() . "\n" );
		}
		echo "Question Set Retrieval (after the fact; works in v4 API): <xmp>";
		var_dump($QuestionSetRetrieve);
		echo "</xmp>\n";
	}
}

// Candidate creation
$name['first'] = 'John';
$name['middle'] = 'W';
$name['last'] = 'Smith';
$dob = '1980-10-10';
$ssn = '123-45-6789';
$passport = ''; // Used for non-US candidates, only
$address['street1'] = '123 Broadway Ave';
$address['street2'] = '';
$address['city'] = 'New York';
$address['state'] = 'NY';
$address['postal_code'] = '10011';
$address['country_code'] = 'US';
$note = 'Test candidate at ' . date("Y-m-d H:i:s") . ' on ' . gethostname();
try {
	$CreateCandidateResult = $blockscore->CreateCandidate($name, $dob, $ssn, $passport, $address, $note);
} catch (Exception $e) {
	die( 'Caught exception: ' .  $e->getMessage() . "\n" );
}

echo "CreateCandidateResult: <xmp>";
var_dump($CreateCandidateResult);
echo "</xmp>\n";

// Candidate edit
if (!empty($CreateCandidateResult['id'])) {
	try {
		$name['first'] = '';
		$name['middle'] = '';
		$name['last'] = '';
		$dob = '';
		$ssn = '987-65-4321';
		$passport = ''; // Used for non-US candidates, only
		$address['street1'] = '';
		$address['street2'] = '';
		$address['city'] = '';
		$address['state'] = '';
		$address['postal_code'] = '';
		$address['country_code'] = '';
		$EditCandidateResult = $blockscore->EditCandidate($CreateCandidateResult['id'], $name, $dob, $ssn, $passport, $address, $note);
	} catch (Exception $e) {
		die( 'Caught exception: ' .  $e->getMessage() . "\n" );
	}
	echo "EditCandidateResult: <xmp>";
	var_dump($EditCandidateResult);
	echo "</xmp>\n";
}

// Search watchlists
if (!empty($CreateCandidateResult['id'])) {
	try {
		$entityType = 'person';
		$SearchWatchlistsResult = $blockscore->SearchWatchlists($CreateCandidateResult['id'], $entityType);
	} catch (Exception $e) {
		die( 'Caught exception: ' .  $e->getMessage() . "\n" );
	}
	echo "SearchWatchlistsResult: <xmp>";
	var_dump($SearchWatchlistsResult);
	echo "</xmp>\n";
}
